cocos/rendering/custom/layout-graph.ts

```diff
@@ -471,11 +473,101 @@ export class DescriptorSetData {
     declare /*refcount*/ descriptorSet: DescriptorSet | null;
 }
 
+export class DescriptorGroupBlockData {
+    constructor (
+        type: DescriptorTypeOrder = DescriptorTypeOrder.UNIFORM_BUFFER,
+        visibility: ShaderStageFlagBit = ShaderStageFlagBit.NONE,
+        accessType: AccessType = AccessType.READ,
+        viewDimension: ViewDimension = ViewDimension.TEX2D,
+        format: Format = Format.UNKNOWN,
+        capacity = 0,
+    ) {
+        this.type = type;
+        this.visibility = visibility;
+        this.accessType = accessType;
+        this.viewDimension = viewDimension;
+        this.format = format;
+        this.capacity = capacity;
```

Comment:
Could we invoke `reset` method in constructor? If we could, it will save some package size.

Reply:
It is possible.

Reply:
In this pr, I will only move classes from `types.ts` to `layout-graph.ts`. `reset` will be changed in the following pr.

Reply:
I found implementations of `constructor` and `reset` are a bit different. For containers, `constructor` will assign to members while `reset` will only change the size to 0.